location_extractor/clients.py

```diff
@@ -0,0 +1,252 @@
+import os
+import sqlite3
+from dataclasses import dataclass
+from sqlite3 import Cursor, Connection
+from typing import List, Tuple, Union, Optional
+
+from location_extractor import src_dir
+from location_extractor.helpers import remove_accents
+import csv
+
+
+@dataclass(frozen=True, order=True)
+class LocationDTO:
+    locale_code: str
+    continent_code: str
+    continent_name: str
+    country_iso_code: str
+    country_name: str
+    subdivision_name: str
+    city_name: str
+    is_in_european_union: bool
+
+
+class DBClient:
+    def __init__(self) -> None:
+        self.dbpath = os.path.join(src_dir, "data", "data.db")
+
+        self.index_columns = [
+            'continent_name',
+            'country_iso_code',
+            'country_name',
+            'subdivision_name',
+            'city_name',
+        ]
+
+        with self.connect() as conn:
+            table_exists = conn.execute("""
+                SELECT name FROM sqlite_master WHERE type='table' AND name='locations';
+            """).fetchone()
+
+            if not table_exists:
+                conn.execute('''
+                    CREATE TABLE locations
+                    (
+                        locale_code text,
+                        continent_code text,
+                        continent_name text,
+                        country_iso_code text,
+                        country_name text,
+                        subdivision_name text,
+                        city_name text,
+                        is_in_european_union boolean,
+                        locale_code_lowercase text,
+                        continent_code_lowercase text,
+                        continent_name_lowercase text,
+                        country_iso_code_lowercase text,
+                        country_name_lowercase text,
+                        subdivision_name_lowercase text,
+                        city_name_lowercase text
+                    )
+                    ''')
+
+                for col in self.index_columns:
+                    conn.execute(f'''
+                        CREATE INDEX {col} 
+                        ON locations({col}_lowercase);
+                    ''')
+
+                locations_path = os.path.join(
+                    src_dir, "data", "GeoLite2-City-CSV_20200303", "GeoLite2-City-Locations-en-processed.csv")
```

Comment:
this formatting is really strange for me, I know it's probably auto formatting of pycharm or your IDE, but what do you think about using following style of formatting multiline calls:

 ```suggestion
                locations_path = os.path.join(
                    src_dir, 
                    "data", 
                    "GeoLite2-City-CSV_20200303", 
                    "GeoLite2-City-Locations-en-processed.csv",
                )
```

Reply:
what about shorter lines like line 26?
` self.dbpath = os.path.join(src_dir, "data", "data.db")`

Reply:
if it fits into 79/80 chars it can be written in one line